### PR TITLE
Added ability to remember previous search, and highlight search field…

### PR DESCRIPTION
**Pull request by MikeWalz11**
… on popup open

tickets: https://github.com/brandon1024/find/issues/40  https://github.com/brandon1024/find/issues/37

Duplicate issues ^^

### CODE REVIEW

**Comment by brandon1024** [comment]
I've made a simplified version of `popup.js`, with some suggestions on ways we could keep the logic modular and open for features in the future, while keeping the logic concise and keeping separation of concerns. Let me know what you think :)

```
window.onload = function() {
    document.getElementById('searchField').addEventListener('keyup', updateLocalStorage);
    retrieveLastSearch(); //Add this here, once all the events are loaded and everything is gucci.
};

function updateLocalStorage() {
    //Create a JSON payload to send to local storage:
    var payload = {'key': 'value'};
    storeDataToLocalStorage(payload);
}

function storeDataToLocalStorage(payload) {
    //only a single method used to store data. Could also create a helper method to
    //update a value in storage.
    chrome.storage.sync.set({payload: payload}, function callback() {
        //LOGGING
    });
}

function retrieveLastSearch() {
    chrome.storage.sync.get('payload', function(data) {
        handleDataFromStorage(data)
    });
}

function handleDataFromStorage(data) {
    //Allows us to manage data in the future all in one location. Keeps this cohesive
    var storagePayload = data.payload;
    
    //grab data we want from payload...
    var previousSearchText = storagePayload.previousSearch;
    
    //do stuff with data...
    changeSearchFieldText(previousSearchText);
}

function changeSearchFieldText(text) {
    //updates to DOM should be done in a single method, to keep the 'view' stuff separate from
    //other logic stuff
    document.getElementById('searchField').value = text;
    document.getElementById("searchField").select();
}

//...
```

**Comment by brandon1024** [comment]
Do you think we should be storing the previous search in the cloud rather than local storage? I've been thinking about it, and I think storing it locally would be a better option, and here is my reasoning:

If I had the extension installed on multiple computers (maybe one of which was a work pc) and happened to be signed into Chrome on both, it makes sense to keep the search history separate because keeping the previous query for your work on your work machine would be more useful.

If at home I were to look for, say, a name to find posts by a person in a Facebook feed, it wouldn't be useful to keep that at work.

What do you think?

**Comment by MikeWalz11** [comment]
@brandon1024 Yea sounds good.  pretty sure its as simple as changing chrome.storage.sync to chrome.storage.local

**Comment by brandon1024** [comment]
I think so too, pretty sure its just a matter of changing `sync` to `local`.

**Comment by brandon1024** [comment]
Testing now :arrows_clockwise:

**Comment by brandon1024** [comment]
Testing Steps:
- :white_check_mark: When opening the extension with no text in storage, loads up good:
image

- :white_check_mark: When opening the extension with text in storage, loads up, shows the previous search, and selects the text in the field:
image

- :white_check_mark: When changing the text from the previous search and closing the extension, on reopening the extension the new text is displayed:
image


@MikeWalz11 Changes look awesome, but looking into it a bit more I think we should update the local storage when the extension closes, not on every keypress. Once this is fixed, I will merge it in 💯

File: popup/popup.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 window.onload = function() {
     document.getElementById('closeBtn').onclick = closeExtension;
-    document.getElementById('searchField').addEventListener('keyup', updateLocalStorage);
+    document.getElementById('searchField').addEventListener('input', updateLocalStorage);
```
Comment (brandon1024) [inline popup/popup.js]: I believe we should change `input` to `change`. This stops it from updating the local storage with every added character. It also has all the correct behaviours, like when closing the extension everything saves correctly.

Comment (MikeWalz11) [inline popup/popup.js]: From testing I noticed that this does work well when closing application through close button, but when using keyboard shortcut the changes are not stored.

Comment (brandon1024) [inline popup/popup.js]: Oh I see this now. Strange, it only had this behavior initially for me. Input event works better, so we may just need to sacrifice a bit of performance. Can update later if we find a solution. Merging.

**Pull request merged**